### PR TITLE
Initial Function: Build frames and Structure

### PR DESCRIPTION
**Pull request by alerrian**
- Build out initial file structure
  - Includes directory for Modules
  - Includes test files and a helper to make writing tests a little easier
  - Includes lib structure and files
- Added class/test interactions
  - Added class for Teams and Games and their associated tests

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    simplecov-html (0.10.2)
+
+PLATFORMS
+  ruby
+
+DEPENDENCIES
+  simplecov
+
+BUNDLED WITH
+   2.0.2
```
Comment (iEv0lv3) [inline Gemfile.lock:19]: I think we can simplify this Gemfile for now with just SimpleCov. Everything else is standard, and we can call it with fewer lines of code. We can always add things as we need them!

File: lib/games.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Games
+  def initialize(game_info)
+    
+  end
+end
```
Comment (iEv0lv3) [inline lib/games.rb]: Re: Style Guide, always leave a newline at the end of files.

File: test/test_helper.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'simplecov'
```
[the remark this answers is not in the document]
Comment (iEv0lv3) [reply test/test_helper.rb:1]: Ignore these for now, and we'll look into it tomorrow.

File: lib/teams.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+              :abbreviation,
+              :stadium
+
+  def initialize(team_data)
+    @franchise_id = team_data[:franchiseId]
+    @team_name = team_data[:teamName]
+    @abbreviation = team_data[:abbreviation]
+    @stadium = team_data[:stadium]
+  end
+end
```
Comment (cgaddis36) [inline lib/teams.rb]: Good work on building out the dummy files and sample tests for Teams/Games

File: test/test_helper.rb [hunk 5]
```diff
@@ -0,0 +1,5 @@
...
+SimpleCov.start
+
+require 'minitest/autorun'
+require 'minitest/pride'
```
[the remark this answers is not in the document]
Comment (iEv0lv3) [reply test/test_helper.rb:5]: Style guide for newline at the end of files is one I'd like to adhere to.

File: test/teams_test.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    assert_instance_of Teams, @team
+  end
+
+  def test_a_team_has_attributes
+    assert_equal 23, @team.franchise_id
+    assert_equal "Atlanta United", @team.team_name
+    assert_equal "ATL", @team.abbreviation
+    assert_equal "Mercedes-Benz Stadium", @team.stadium
+  end
+end
```
Comment (iEv0lv3) [inline test/teams_test.rb]: Same thing re: new line at end of files.

File: test/teams_test.rb [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+  def test_a_team_exists
+    assert_instance_of Teams, @team
+  end
+
+  def test_a_team_has_attributes
+    assert_equal 23, @team.franchise_id
+    assert_equal "Atlanta United", @team.team_name
+    assert_equal "ATL", @team.abbreviation
+    assert_equal "Mercedes-Benz Stadium", @team.stadium
```
Comment (iEv0lv3) [inline test/teams_test.rb]: Single quotes as the default unless double are necessary for string interpolation.

File: test/teams_test.rb [hunk 8]
```diff
@@ -0,0 +1,24 @@
...
+  end
+
+  def test_a_team_exists
+    assert_instance_of Teams, @team
+  end
+
+  def test_a_team_has_attributes
+    assert_equal 23, @team.franchise_id
+    assert_equal "Atlanta United", @team.team_name
+    assert_equal "ATL", @team.abbreviation
```
Comment (iEv0lv3) [inline test/teams_test.rb]: Same thing here for single quotes.

File: test/teams_test.rb [hunk 9]
```diff
@@ -0,0 +1,24 @@
+require_relative 'test_helper'
+require_relative '../lib/teams'
+
+class TeamsTest < Minitest::Test
+  def setup
+    @team = Teams.new({
```
Comment (iEv0lv3) [inline test/teams_test.rb]: Fuck this. We can change the settings. Parentheses is the way to go for arguments.

File: test/teams_test.rb [hunk 10]
```diff
@@ -0,0 +1,24 @@
+require_relative 'test_helper'
```
Comment (iEv0lv3) [inline test/teams_test.rb]: Ignore.

File: test/stat_tracker_test.rb [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+    game_teams_path = './data/game_teams.csv'
+
+    locations = {
+                  games: game_path,
+                  teams: team_path,
+                  game_teams: game_teams_path
+                }
+
+    stat_tracker = StatTracker.from_csv(locations)
+    game = ["2012030221", "20122013", "Postseason", "5/16/13", "3", "6", "2", "3", "Toyota Stadium", "/api/v1/venues/null"]
```
[the remark this answers is not in the document]
Comment (iEv0lv3) [reply test/stat_tracker_test.rb:26]: Agree with Hound.

File: test/stat_tracker_test.rb [hunk 12]
```diff
@@ -0,0 +1,30 @@
...
+  def test_stat_tracker_from_csv
+    game_path = './data/games.csv'
+    team_path = './data/teams.csv'
+    game_teams_path = './data/game_teams.csv'
+
+    locations = {
+                  games: game_path,
+                  teams: team_path,
+                  game_teams: game_teams_path
+                }
```
[the remark this answers is not in the document]
Comment (iEv0lv3) [reply test/stat_tracker_test.rb:23]: Agree with Hound.

File: test/stat_tracker_test.rb [hunk 13]
```diff
@@ -0,0 +1,30 @@
...
+    assert_instance_of StatTracker, new_tracker
+  end
+
+  def test_stat_tracker_from_csv
+    game_path = './data/games.csv'
+    team_path = './data/teams.csv'
+    game_teams_path = './data/game_teams.csv'
+
+    locations = {
+                  games: game_path,
```
[the remark this answers is not in the document]
Comment (iEv0lv3) [reply test/stat_tracker_test.rb:20]: Agree with Hound.

**Comment by iEv0lv3** [comment]
Overall looks really good! Thanks for working on these. Ignore the Hound comments I suggested, and we'll work on getting it setup to our liking. After we merge the runner branch, pull that down locally, merge into your local branch and resolve any conflicts, and then push and PR back into dev and we'll get it merged. This should avoid most conflicts.

**Comment by cgaddis36** [comment]
Good work, looks like we have a solid framework to start building out these complex methods from.

**Review by iEv0lv3 — APPROVED** [review]
This PR has been reviewed line by line, and it looks like there won't be merge conflicts. We'll give it a try and see how it goes!

**Pull request merged**